### PR TITLE
Add rev 1 of Kafka notifier

### PR DESCRIPTION
**Pull request by MrCreosote**
Only saveObjects implemented for now.

TODO:
* Add more implementations
  * Will require adding user to method sigs in most cases
* Document how to configure the notifier

### CODE REVIEW

File: src/us/kbase/workspace/modules/KafkaNotifierFactory.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+ * The listener is configured to require a full write to all the replicates before a call
+ * to Kafka returns, and if a write fails, an exception is thrown in the thread that called
+ * the listener.
+ * 
+ * @author gaprice@lbl.gov
+ *
+ */
+public class KafkaNotifierFactory implements WorkspaceEventListenerFactory {
+
+		// TODO DOCS
```
Comment (sychan) [inline src/us/kbase/workspace/modules/KafkaNotifierFactory.java]: Some docs about the JSON object that you are pushing into kafka would be very helpful - even just for a reviewer of your PR.

Comment (MrCreosote) [reply src/us/kbase/workspace/modules/KafkaNotifierFactory.java]: I'm not going to write any docs until I have the JSON object figured out for all the required events, but if you have any specific questions I can answer them.

The current format looks like this:
```
{"wsid":1,"ver":1,"evtype":"NEW_VERSION","objid":16,"time":1547754188639,"objtype":"Empty.AType-1.0","user":"gaprice"}
```

Comment (sychan) [reply src/us/kbase/workspace/modules/KafkaNotifierFactory.java]: It doesn't need to be a formal document. Just comments in the code would be good. I think that Shane, Bill and yourself are all working on kafka messages, putting a comment in the code that just shows the current state of the JSON object would make it easier for all of you to peek over each other's shoulders and see where things are headed.

Comment (MrCreosote) [reply src/us/kbase/workspace/modules/KafkaNotifierFactory.java]: This is going to be done today or tomorrow and I'll document it at that point, and it's documented above. It's WIP mang

Comment (MrCreosote) [reply src/us/kbase/workspace/modules/KafkaNotifierFactory.java]: Alternatively, here's the documentation in the code: https://github.com/kbase/workspace_deluxe/pull/225/files#diff-2d3262cb67db8004f2545742863b5e42R285

**Pull request merged**